src/dfmt/ast.d

```diff
@@ -0,0 +1,3926 @@
+module dfmt.ast;
+
+import dmd.visitor;
+import dmd.astcodegen;
+import dmd.astenums;
+import dmd.tokens;
+import dmd.parse;
+import dmd.mtype;
+import dmd.identifier;
+import dmd.hdrgen;
+import dmd.init;
+import dmd.rootobject;
+import dmd.target;
+import dmd.root.string : toDString;
+import dmd.optimize : optimize;
```

Comment:
```suggestion

```
looks like it can be zapped

Reply:
It's needed for calling `Expression.optimize`, which was moved out of `expression.d` into `optimize.d` in a recent PR. It breaks CI because I don't think there's been a release with the change yet, I compile dmdfmt with the master branch of dmd locally so I'm sometimes ahead of the CI pipeline. I'll update `dub.json` to use [v2.107.0-beta.1](https://github.com/dlang/dmd/releases/tag/v2.107.0-beta.1), that should fix it.

Reply:
and where

Reply:
It's called [here](https://github.com/snprajwal/dfmt/blob/saoc-2023/src/dfmt/ast.d#L2211) and [here](https://github.com/snprajwal/dfmt/blob/saoc-2023/src/dfmt/ast.d#L2282). It was introduced to fix https://issues.dlang.org/show_bug.cgi?id=7375, and was taken from `hdrgen.d` in DMD. Now that I look closer at it, it's a compile time check, so shouldn't be needed in dfmt. I'll get rid of it.